sys-boot/gnu-efi: Add custom-cflags flag and prevent avx instructions

BinaryKhaos opened the pull request:
Building gnu-efi with custom compiler flags is risky at best and should generally be avoided.

Nevertheless, if custom flags are used, we need to make sure no avx instructions are generated as those cause a non-functional gnu-efi build. This is only required for >= 3.0.5 as prior to this, "-mno-sse" was used which also implied no avx.

Fixes bug #619628.

floppym commented on sys-boot/gnu-efi/gnu-efi-3.0.6-r1.ebuild:
```diff
@@ -29,6 +29,18 @@ RDEPEND=""
+pkg_pretend() {
+	if use custom-cflags; then
+		ewarn
+		ewarn "You have enabled building with USE=custom-cflags. Be aware that"
```
This message is a bit... wordy, and i don't like introducing pkg_pretend just for this.

Could you just mask the USE flag in profiles/base/package.use.mask instead?

BinaryKhaos replied:
Sure. I actually think that is way better. I would recommended doing the same for the rEFInd ebuild as well. I copied that msg from there, by the way, with slight modifications. :) If you don't mind, I would adapt both and update the pull request appropriately tomorrow.

floppym commented on sys-boot/gnu-efi/gnu-efi-3.0.6-r1.ebuild:
```diff
@@ -61,8 +73,15 @@ efimake() {
+		filter-mfpmath sse
+
+		# https://bugs.gentoo.org/619628
+		append-flags -mno-avx
```
This will break on non-x86 platforms, like ARM and IA64.
```
% aarch64-unknown-linux-gnu-gcc -mno-avx
aarch64-unknown-linux-gnu-gcc: error: unrecognized command line option ‘-mno-avx’; did you mean ‘-Wno-a,’?
```
Maybe replace it with this:
```
append-cflags $(test-flags-CC -mno-avx)
```

BinaryKhaos replied:
I should have known this. Sorry. Just for future reference: Do you have several test installations were you test those things? I was just wondering what the usual procedure is in cases of different architectures were things could essentially be different and/or problematic.

floppym replied:
I just happened to catch this when reading the diff. I only have amd64 systems at my disposal, but I also have arm/arm64 cross-compilers installed for testing at build time.

BinaryKhaos commented:
Ok, I have pushed an amended commit which should address your comments. One thing that is debatable in this commit is that I have also added sys-boot/refind to package.use.mask since it depends on gnu-efi and basically has the same requirements and problems when it comes to compile flags... but this addition is a bit out of place for the topic of this commit, I agree. One the other hand, it was a natural fit and I wanted to avoid to open another pull request just for this one line addition... which would depend on a pending pull request.

BinaryKhaos commented:
Ok, I have updated the branch and addressed your comments. I hope everything is fine, otherwise I would appreciate any comments you might have. Thanks for taking the time, by the way!

floppym commented:
Rebased as ed5268032c9102500d64d011c4c308489e5266d2. Thanks!
